src/components/course/CourseRunSelector.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext, useState, useRef } from 'react';
+import moment from 'moment';
+
+import { CourseContext } from './CourseContextProvider';
+import { SET_COURSE_RUN } from './data/constants';
+
+import './styles/CourseRunSelector.scss';
+
+export default function CourseRunSelector() {
+  const { dispatch, state } = useContext(CourseContext);
+  const {
+    activeCourseRun,
+    course: {
+      courseRuns,
+    },
+  } = state;
+  const [editing, setEditing] = useState(false);
+  const selectRef = useRef(activeCourseRun.uuid);
+  const multipleRunsAvailable = courseRuns && courseRuns.length > 1;
```

Comment:
NONBLOCKING SUGGESTION: If you are interested, I think it is possible to pull out these pieces into a separate hook which can then (hopefully easily) tested with the react hooks testing library, for better testability. The output of that hook would be multipleRunsAvailable, I believe... the input would be a a fake context or sth like that

even if not for this PR I thikn that may be an interesting pattern to use in tests that invoke hook level code .. 

Alternatively, a test can be added via enzyme (my future vision is we can migrate to React testing lib), to test that component render does work with a correct multipleRunsAvailable  but not sure what that looks like for this feature

I recently added the lib into the project and if you are curios here is a test that uses the hooks testing lib

https://github.com/edx/frontend-app-learner-portal-enterprise/blob/master/src/components/enterprise-page/data/tests/customerConfig.test.js

looks great otherwise!

hth

Reply:
For some one who's new to hooks, what benefit does this have by making it a hook instead of a locally scoped variable? For context, the only reason I have it declared at all is for readability purposes as to what I'm conditionalizing, as otherwise I'd just reference the state I have (no transforms or anything needed).

 I'd imagine testing would be based on whatever state we decide to seed, success case receives an array with > 1 element, failure receive 1 or 0.

Reply:
For testing `CourseRunSelector`, I think we'll be able to wrap `<CourseRunSelector />` inside of a fake `CourseContext`, once where there's only 1 course run and again where there's 2+ course runs. In the former, `CourseRunSelector` would not render anything; in the latter, `CourseRunSelector` would be rendered.

I don't see an immediate need to abstract that logic out into a hook quite yet, unless that logic was used elsewhere within the page/app but as far as I know/remember, this is only place that deals with looking at the course runs right now.